AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -239,7 +284,14 @@ internal class CardViewController: FormViewController {
     
     private func didReceived(bin: String) {
         self.securityCodeItem.selectedCard = supportedCardTypes.adyen.type(forCardNumber: bin)
-        throttler.throttle { [weak self] in self?.cardDelegate?.didChangeBIN(bin) }
+        let shouldShow = configuration.showsKoreanAuthentication
+        throttler.throttle { [weak self] in
+            self?.cardDelegate?.didChangeBIN(bin)
+
+            let binIsLong = bin.count >= BinInfoProvider.minBinLength
+            self?.additionalAuthPasswordItem.isHidden.wrappedValue.setTrueUnless(binIsLong && shouldShow)
+            self?.additionalAuthCodeItem.isHidden.wrappedValue.setTrueUnless(binIsLong && shouldShow)
```

Comment:
maybe extract the content of the block into a function? so that you  do `self?` once.

Reply:
👍